examples/tutorials/mnist_pytorch/train.py

```diff
@@ -80,7 +80,6 @@ def evaluate_batch(self, batch: pytorch.TorchData, batch_idx: int) -> Dict[str,
         return {
             "validation_loss": validation_loss,
             "accuracy": accuracy,
-            "batch": self.context.current_train_batch(),
```

Comment:
I actually like making the user emit the special metrics they care about.  It is not difficult.

If we are going to do it, I feel pretty strongly that we need to stick to `steps_completed` as the name for batches, since that's basically everywhere in our system, and I don't want to introduce `batch` as an off-by-one alias of `steps_completed` soley for the purpose of having `time_unit: batch` look nicer than `time_unit: steps_completed`.

I'm open to either option.

Reply:
my reasoning for not making the user do it:
- this is a managed training loop, user isn't calling `core.train.report_metrics` themselves. there's some expectation that it'll "just work".
- we don't support anything besides "batch"/"epoch" for `time_metric` in PyTorchTrial/Trainer anyway. so it's not that the user cares about these "batch"/"epoch" metrics, it's that we do. 

my reasoning for making the user do it:
- >It is not difficult.
- we already make them do this if they want "epoch" behavior in the web UI. 
- the validation metrics aren't arbitrary as is, you must report the searcher metric yourself which is also configured in the expconf, and this feels similar. 
- it's explicit about what the system actually expects, no magic.

i can see both ways as well, but "no magic" is pretty compelling to me, and i _really_ don't want to do `time_unit: steps_completed`, so i changed it back to "user must report" behavior.

Reply:
We talked out-of-band, and agreed that:
- We like the simplicity of "you must emit the time_metric" you want for the searcher to work
- We also like the ease-of-use of just reporting "epochs" and "batches" automatically, which additionally supports the epochs-axis in the webui (with some small backend changes)
- We don't like magically including extra metrics based on what the time_metric is.  "epochs" and "batches" should just be emitted unconditionally.